src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig

```diff
@@ -71,51 +91,61 @@
 
 {% block checkbox_widget -%}
     {%- set parent_label_class = parent_label_class|default(label_attr.class|default('')) -%}
-    {%- set attr = attr|merge({class: (attr.class|default('') ~ ' form-check-input')|trim}) -%}
-    {% if 'checkbox-inline' in parent_label_class %}
-        {{- form_label(form, null, { widget: parent() }) -}}
-    {% elseif 'form-check-inline' in parent_label_class %}
-        <div class="form-check{{ not valid ? ' form-control is-invalid' }} form-check-inline">
+    {%- if 'checkbox-custom' in parent_label_class -%}
+        {%- set attr = attr|merge({class: (attr.class|default('') ~ ' custom-control-input')|trim}) -%}
+        <div class="custom-control custom-checkbox{{ 'checkbox-inline' in parent_label_class ? ' custom-control-inline' }}">
             {{- form_label(form, null, { widget: parent() }) -}}
+            {{- form_errors(form) -}}
         </div>
-    {% else -%}
-        <div class="form-check{{ not valid ? ' form-control is-invalid' }}">
+    {%- else -%}
+        {%- set attr = attr|merge({class: (attr.class|default('') ~ ' form-check-input')|trim}) -%}
+        <div class="form-check{{ 'checkbox-inline' in parent_label_class ? ' form-check-inline' }}">
             {{- form_label(form, null, { widget: parent() }) -}}
+            {{- form_errors(form) -}}
         </div>
     {%- endif -%}
 {%- endblock checkbox_widget %}
 
 {% block radio_widget -%}
     {%- set parent_label_class = parent_label_class|default(label_attr.class|default('')) -%}
-    {%- set attr = attr|merge({class: (attr.class|default('') ~ ' form-check-input')|trim}) -%}
-    {%- if 'radio-inline' in parent_label_class -%}
-        {{- form_label(form, null, { widget: parent() }) -}}
+    {%- if 'radio-custom' in parent_label_class -%}
+        {%- set attr = attr|merge({class: (attr.class|default('') ~ ' custom-control-input')|trim}) -%}
+        <div class="custom-control custom-radio{{ 'radio-inline' in parent_label_class ? ' custom-control-inline' }}">
+            {{- form_label(form, null, { widget: parent() }) -}}
+            {{- form_errors(form) -}}
+        </div>
     {%- else -%}
-        <div class="form-check{{ not valid ? ' form-control is-invalid' }}">
+        {%- set attr = attr|merge({class: (attr.class|default('') ~ ' form-check-input')|trim}) -%}
+        <div class="form-check{{ 'radio-inline' in parent_label_class ? ' form-check-inline' }}">
             {{- form_label(form, null, { widget: parent() }) -}}
```

Comment:
I think you want to do 
```
{{- form_label(form, null) -}}
{{- parent() -}}
```
here, because in [Bootstrap 4 Labels are not supposed](https://getbootstrap.com/docs/4.0/migration/#forms) to wrap [checkboxes](https://getbootstrap.com/docs/4.0/components/forms/#default-stacked) anymore. (Used to make the label text gray when the checkbox is disabled)

Same thing in line 114 above.

Reply:
FYI: 
One could improve the bootstrap way.. Here is a non-bootstrap example how one use checkboxes/radios. 

```html

<fieldset>
  <legend>Sex</legend>
    <label for="male">
    <input type="radio" name="sex" id="male">Male</label>
    <label for="female">
    <input type="radio" name="sex" id="female">Female</label>
</fieldset>

```

Reply:
@Nyholm just for your consideration for future comments. Let's never use Sex or Gender in any example or comment. It's difficult to use them right without offending someone. You can use examples like this as an alternative. Thanks!

```html
<fieldset>
    <legend>Delivery</legend>
    <label for="standard">
    <input type="radio" name="delivery" id="standard">Standard</label>
    <label for="priority">
    <input type="radio" name="delivery" id="priority">Priority</label>
</fieldset>
```

Reply:
Sorry you are correct, I just copied code from a WCAG2 review that I had lying around.